code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -378,6 +380,45 @@
 	anchored = FALSE
 	target = null
 
+
+/mob/living/simple_animal/bot/cleanbot/larry/attackby(obj/item/I, mob/living/user)
+	if(I) //Does the arg exist?
+		if(istype(I, /obj/item/kitchen/knife)) //Is it a knife?
+			if(!knife)
+				var/obj/item/kitchen/knife/newknife = I
+				knife = newknife
+				newknife.forceMove(src)
+				message_admins("[user] attached a [newknife.name] to [src]") //This should definitely be a notified thing.
+				src.AddComponent(/datum/component/larryknife, knife.force)
+				update_icons()
+	else
+		return ..()
+
+/mob/living/simple_animal/bot/cleanbot/larry/update_icons()
+	if(knife)
+		var/mutable_appearance/knife_overlay = knife.build_worn_icon(state = knife.icon_state, default_layer = 20, default_icon_file = 'icons/mob/inhands/misc/larry.dmi')
+		add_overlay(knife_overlay)
```

Comment:
If you are adding overlays, don't you need to first cut overlays to prevent them stacking up infinitely?

Reply:
Not sure, but I also don't think cleanbots or parent types have any overlay stuff